From 636f8de7b5cd826abaec7dcea85d530e6afb4701 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Thu, 4 Feb 2010 13:48:59 +0800
Subject: [PATCH 096/280] ENGR00120773 Ipu screenlayer lib: close shm file
 descriptor after use

CreateScreenLayer() too many times within a process will cause file open
fail.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 screenlayer/ScreenLayer.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/screenlayer/ScreenLayer.c b/screenlayer/ScreenLayer.c
index 7e3594c..2708428 100644
--- a/screenlayer/ScreenLayer.c
+++ b/screenlayer/ScreenLayer.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2009-2010 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  */
 
@@ -655,11 +655,13 @@ SLRetCode PreScreenLayerIPC(char *pFbdev)
 		sem_close(semMainID);
 		sem_close(semLoadID);
 		sem_close(semDispID);
-		goto pre_err0;
+		goto pre_err1;
 	}
 
 	sem_post(semMainID);
 	dbg(DBG_DEBUG, "PreScreenLayerIPC end!\n");
+pre_err1:
+	close(shmID);
 pre_err0:
 	return ret;
 }
-- 
1.8.0

